hooks/addon

```diff
@@ -218,6 +222,51 @@ fly)
   exit $?
   ;;
 
+setup-approle)
+  # If AppRole is already enabled in this Vault, it'll return a 400 error with "path already in use". This is OK.
+  if [[ ! $(safe vault auth enable approle 2>&1) =~ (Success\! Enabled approle auth method)|(path is already in use) ]] ; then
+    echo "Failed to enable AppRole on your targeted Vault. Womp womp :("
+    exit 1
+  fi
+
+  if ! safe exists "secret/exodus/$GENESIS_ENVIRONMENT/concourse" ; then
+    error "#R{[ERROR]} Cannot find exodus data for #C{$GENESIS_ENVIRONMENT}.  Please redeploy" \
+          "        before using addons."
+  fi
+
+  # Create AppRole Role named "genesis-piplines" with the following parameters:
+  # - 90 minute TTL (Some BOSH deployments could take awhile, and then the Exodus write-back could fail if TTL is too short)
+  # - unlimited token use
+  safe --quiet set auth/approle/role/genesis-pipelines \
+    secret_id_ttl=90m \
+    token_num_uses=0 \
+    token_ttl=90m \
+    token_max_ttl=90m \
+    secret_id_num_uses=0
+  role_id=$(safe get auth/approle/role/genesis-pipelines/role-id:role_id)
+  # generate a secret key for the newly minted role
+  approle_secret=$(safe vault write -field=secret_id -f auth/approle/role/genesis-pipelines/secret-id)
+
+
+  # write info to exodus
+  safe --quiet set secret/exodus/ci/genesis_pipelines approle_id=$role_id
+  safe --quiet set secret/exodus/ci/genesis_pipelines approle_secret=$approle_secret
```

Comment:
Can we match the hyphens in the app role here?  i.e. `secret/exodus/ci/genesis-pipelines`?